File: core/src/test/java/me/loki2302/AppTest.java
package me.loki2302;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class AppTest {
    @Test
    public void canRunApp() {
        App.main(null);
    }

    @Test
    public void canUseCalculator() {
        Calculator calculator = new Calculator();
        assertEquals(5, calculator.add(2, 3));
    }

    @Test
    public void canUseStatefulCalculator() {
        StatefulCalculator statefulCalculator = new StatefulCalculator(2);
        assertEquals(5, statefulCalculator.add(3));
    }
}

File: learning-scala/spring-app/src/test/scala/me/loki2302/springapp/AppTest.scala
package me.loki2302.springapp

import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.boot.test.{SpringApplicationConfiguration, IntegrationTest}
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
import org.springframework.test.context.web.WebAppConfiguration
import org.springframework.web.client.RestTemplate
import org.junit.Assert._

@RunWith(classOf[SpringJUnit4ClassRunner])
@SpringApplicationConfiguration(classes = Array(classOf[Config]))
@WebAppConfiguration
@IntegrationTest
class AppTest {
  @Test
  def dummy() : Unit = {
    val restTemplate: RestTemplate = new RestTemplate
    val result = restTemplate.getForObject("http://localhost:8080/", classOf[String])
    assertEquals("hello", result)
  }
}
